pkg/cache/ristretto/ristretto.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ristretto
+
+import (
+	"context"
+	"encoding/json"
+	"sync"
+	"time"
+
+	"github.com/cespare/xxhash/v2"
+	"github.com/deislabs/ratify/pkg/cache"
+	"github.com/dgraph-io/ristretto"
+	"github.com/dgraph-io/ristretto/z"
+	"github.com/sirupsen/logrus"
+)
+
+type ristrettoFactory struct {
+	once sync.Once
+}
+
+type ristrettoCache struct {
+	memoryCache *ristretto.Cache
+}
+
+func init() {
+	cache.Register("ristretto", &ristrettoFactory{})
+}
+
+func (f *ristrettoFactory) Create(ctx context.Context, cacheEndpoint string, cacheSize int) (cache.CacheProvider, error) {
+	var err error
+	var memoryCache *ristretto.Cache
+	f.once.Do(func() {
+		memoryCache, err = ristretto.NewCache(&ristretto.Config{
```

Comment:
Does the ristretto implementation check if host can actually support upper bound?